### DISCUSSION TITLE
Support for Front Matter

### DISCUSSION THREAD

**Discussion by Kariton** [bowman2001/perplex#167] [Ideas]
I case you dont know:
[Frontmatter](https://frontmatter.codes) describes itself as:
> Headless CMS right in your code editor
that helps managing your static sites and Markdown based sites. Supports Hugo, Jekyll, Docusaurus, NextJs, Gatsby, and more.
Experience the best of both worlds with Front Matter, the CMS that runs within Visual Studio Code, GitPod, and many more. Get the control over your static generator's content by bringing in the CMS features from Front Matter. Start creating and previewing content in real-time - no complicated setup - choose Front Matter CMS and start creating amazing content today.

because management tools makes the life simpler i have tried this one with your template.
and i like it.

the installation and usage is straight forward.
- install the plugin
- follow the guided initial setup

the configuration is a bit more complex for this template.
the plugin depends on a json that is a write-down of all metadata fields. basically a schema file.

i have creating a PoC `frontmatter.json` (attachment) file which tells VSCode / the plugin how thinks work.
[frontmatter.json](https://github.com/bowman2001/perplex/files/14983466/frontmatter.json)

this needs to be inside the project root.

after that there are UI elements within VS with suggestions for tags or categories, mandatory fields and field type validation.

maybe you want give it a try.

**Comment by Kariton** [comment]
for front matter to work reliably every already existing markdown file does need a new value:
`type: {{type}}`

where 'type' is the corresponding `frontmatter.json` definition

**Comment by bowman2001** [comment]
Yes, Frontmatter for VS Code is good tool. But it relies on a central media repository which is not completely compatible with all the options the theme offers for images. 

I haven't tried recently, but it should work with the theme if you don't use the enhanced layout options and image captions. 

See <https://perplex.desider.at/doc/intro/markdown/#frontmatter-cms>
